### ISSUE TITLE
"An element's ID must be unique" is not an accessibility check

### ISSUE THREAD

**Issue by calebegg** [GoogleChrome/accessibility-developer-tools#210]
I don't think checking for unique IDs is a good fit for an accessibility audit. We use some SVG icons exported from Adobe Illustrator, which adds in some automatically generated IDs. If those icons occur in the document multiple times, this check fails.

It would certainly be better if we could remove them, but I don't see why this should fail the accessibility audit unless those IDs are referenced as part of an aria-* attribute (they are not).

**Comment by RickSBrown** [comment]
@calebegg a11y is frequently highly impacted by duplicate ids, discussed in much more detail #141

**Comment by RickSBrown** [comment]
@alice not sure how to handle duplicates but this is a duplicate of #141 - do we just close?

**Comment by RickSBrown** [comment]
This one got me thinking... I'm actually surprised that the ids in the SVG are considered by the browser for the purpose of `getElementById`.

I don't think it changes anything but I guess theoretically it would be possible to ignore any elements that are not in an HTML namespace. Of course the browser doesn't do that so I'm not sure it would make sense.

**Comment by wgl85** [comment]
When using polymer reusable web components to build our page, it received many element's ID  must be unique violations (100+). Polymer provides its shadow dom for its elements, even the element is reused multiple times in the same page, there should be no problem of duplicate ID. Is there anyway to avoid this violation? I have tried inserting aria-hidden="true", but failed.

**Comment by alice** [comment]
@wgl85 I would recommend turning this rule off for the time being:
- If you're using the extension, you can do this via the extension options.
- If you're using the library, you can pass an `AuditConfiguration` object with `auditRulesToIgnore = ['duplicateId']`

I would imagine that if you're seeing this issue with Polymer, then you are probably testing it using the "shady dom" polyfill rather than native shadow DOM, since the test uses a query selector which would not select elements inside of true shadow DOM.

It seems like this test is very noisy, sadly. IDREFs in the same scope really should be unique, otherwise there is no point having them, but it is indeed the case that automatically generated code often generates non-unique IDREFs and it is harmless unless something attempts to refer to that ID. Since there is a finite number of situations in which IDREFs may be used, perhaps we should follow @dmazzoni-google's original suggestion of finding all the cases where an IDREF is specified and make sure that each one refers to exactly one element.

**Comment by alice** [comment]
@RickSBrown Yeah, I don't think GH issues have an official mechanism, but I think commenting "dup of #<id>" and closing is fine. I'll do that now, and copy my reasoning above into the other bug.

**Comment by alice** [comment]
Duplicate of #141.
